client/app/hosting/local-seo/hosting-local-seo.controller.js

```diff
@@ -0,0 +1,100 @@
+angular.module('App').controller(
+  'HostingTabLocalSeoCtrl',
+  class HostingTabLocalSeoCtrl {
+    constructor($scope, $stateParams, $q, $window, HostingLocalSeo) {
+      this.$scope = $scope;
+      this.$stateParams = $stateParams;
+      this.$q = $q;
+      this.$window = $window;
+      this.HostingLocalSeo = HostingLocalSeo;
+    }
+
+    $onInit() {
+      this.loading = {
+        accounts: true,
+        locations: false,
+      };
+      this.productId = this.$stateParams.productId;
+
+      this.accounts = null;
+      this.locations = null;
+      this.locationDetails = null;
+
+      this.loadAccountsAndLocations();
+    }
+
+    loadAccountsAndLocations() {
+      this.loading.accounts = true;
+      return this.getAccounts()
+        .then((accounts) => {
+          this.accounts = accounts;
+          if (!_.isEmpty(accounts)) {
+            return this.getLocations().then((locations) => {
+              this.locations = locations;
+            });
+          }
+        })
+        .finally(() => {
+          this.loading.accounts = false;
+        });
+    }
+
+    getAccounts() {
+      return this.HostingLocalSeo.getAccounts(this.productId)
+        .then(accounts => this.$q.all(
+          _.map(accounts, account => this.HostingLocalSeo.getAccount(this.productId, account)),
+        ));
+    }
+
+    getLocations() {
+      return this.HostingLocalSeo.getLocations(this.productId);
+    }
+
+    hasAccounts() {
+      return !_.isEmpty(this.accounts);
+    }
+
+    hasLocations() {
+      return !_.isEmpty(this.locations);
+    }
+
+    transformItem(locationId) {
+      this.loading.locations = true;
+      return this.HostingLocalSeo.getLocation(this.productId, locationId)
+        .then((location) => {
+          const accountId = _.get(location, 'accountId');
+          if (accountId) {
+            const account = _.find(this.accounts, { id: accountId });
+            if (account) {
+              location.account = account;
+            }
+          }
+          return location;
+        });
+    }
+
+    transformItemDone() {
+      this.loading.locations = false;
+    }
+
+    refreshLocations() {
+      this.loadAccountsAndLocations();
+    }
+
+    goToInterface(location) {
+      if (!location.accountId) {
+        return;
+      }
+
+      /*
+        Opening the window first then setting the location prevents browsers
+        from blocking it as a popup.
+      */
+      const win = this.$window.open('', '_blank');
```

Comment:
You also need to add `win.opener = null;` to prevent malicious attacks (using the option "noopener" is a no-go because it makes the function return `null`). Some kind of service for that would be cool.